apis/python/src/tiledbsoma/io/registration/ambient_label_mappings.py

```diff
@@ -0,0 +1,402 @@
+                    raw_var_next_soma_joinid += 1
+
+            var_maps["raw"] = raw_var_map
+
+        tiledbsoma.logging.logger.info(
+            f"Registration: accumulated to nobs={len(obs_map)} nvar={len(var_map)}."
+        )
+        return cls(
+            obs_axis=AxisAmbientLabelMapping(data=obs_map, field_name=obs_field_name),
+            var_axes={
+                ms_name: AxisAmbientLabelMapping(data=data, field_name=var_field_name)
+                for ms_name, data in var_maps.items()
+            },
+        )
+
+    @classmethod
+    def from_h5ad_append_on_experiment(
+        cls,
+        h5ad_file_name: str,
+        previous: Self,
+        *,
+        measurement_name: str,
+        obs_field_name: str = "obs_id",
+        var_field_name: str = "var_id",
+    ) -> Self:
+        """Extends registration data to one more H5AD input file."""
+        tiledbsoma.logging.logger.info(f"Registration: registering {h5ad_file_name}.")
+
+        adata = ad.read_h5ad(h5ad_file_name, "r")
+
+        return cls.from_anndata_append_on_experiment(
+            adata,
+            previous,
+            measurement_name=measurement_name,
+            obs_field_name=obs_field_name,
+            var_field_name=var_field_name,
+        )
+
+    @classmethod
+    def from_h5ad_appends_on_experiment(
+        cls,
+        experiment_uri: Optional[str],
+        h5ad_file_names: Sequence[str],
+        *,
+        measurement_name: str,
+        obs_field_name: str,
+        var_field_name: str,
+    ) -> Self:
+        """Extends registration data from the baseline, already-written SOMA
+        experiment to include multiple H5AD input files."""
+
+        if experiment_uri is not None and tiledbsoma.Experiment.exists(experiment_uri):
+            registration_data = cls.from_isolated_soma_experiment(
+                experiment_uri,
+                obs_field_name=obs_field_name,
+                var_field_name=var_field_name,
+            )
+        else:
+            registration_data = cls(
+                obs_axis=AxisAmbientLabelMapping(data={}, field_name=obs_field_name),
+                var_axes={
+                    measurement_name: AxisAmbientLabelMapping(
+                        data={}, field_name=var_field_name
+                    ),
+                    "raw": AxisAmbientLabelMapping(data={}, field_name=var_field_name),
+                },
+            )
+
+        for h5ad_file_name in h5ad_file_names:
+            registration_data = cls.from_h5ad_append_on_experiment(
+                h5ad_file_name,
+                registration_data,
+                measurement_name=measurement_name,
+            )
+
+        tiledbsoma.logging.logger.info("Registration: complete.")
+        return registration_data
+
+    def show(self) -> None:
+        print(f"obs:{len(self.obs_axis.data)}")
+        for k, v in self.var_axes.items():
+            print(f"{k}/var:{len(v.data)}")
```

Comment:
Would this make sense as a `__str__` or `__repr__` method that instead returns the contents that you’re printing here?